src/hardware/mixer.cpp

```diff
@@ -242,14 +252,31 @@ void MixerChannel::Enable(const bool should_enable)
 	MIXER_UnlockAudioDevice();
 }
 
-void MixerChannel::SetFreq(Bitu freq) {
-	freq_add=(freq<<FREQ_SHIFT)/mixer.freq;
+void MixerChannel::SetFreq(Bitu freq)
+{
+	if (!freq) {
+		// If the channel rate is zero, then avoid resampling by running
+		// the channel at the same rate as the mixer
+		assert(mixer.freq > 0);
```

Comment:
Yes; `[mixer] rate` isn't a free wheeling int.. it can only be one of the fixed 8, 11, 16, 22, 32, 44, 48, and 49 KHz rates in the array. If it's set to something else (like zero), we get:

```
CONFIG: Loading primary file dosbox.conf
"0" is not a valid value for variable: rate.
It might now be reset to the default value: 44100
```